### PR TITLE
Support windows CR characters

### PR DESCRIPTION
**Pull request by AnEmortalKid**
Confirm I can build this now on my windows machine after running `dos2unix`.

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Developers
+
+Notes for developers of the language.
+
+## Build
+
+1. `npm run build`
+  2. Requires files do not contain `\r` (CR) returns and must only include `\n` (LF)
```
Comment (vpzomtrrfrt) [inline DEVELOPERS.md:8]: I assume this is just an issue with the new parser? If so we should just change it to work with either

File: src/bin.djs [hunk 2]
```diff
@@ -3,7 +3,7 @@ so ./index as doge3
 
 very file is 0
 rly process giv argv giv length bigger 2
-    file is process giv argv levl 2
+  file is process giv argv levl 2
```
Comment (vpzomtrrfrt) [inline src/bin.djs:6]: does this change have any significance? 4 spaces is the convention per the 2.x spec document

**Review by vpzomtrrfrt — CHANGES_REQUESTED** [review]

File: src/bin.djs (hunk 2 above)
Comment (AnEmortalKid) [reply src/bin.djs:6]: Ah, no, i just put a `tab` in there on accident when mucking with it and then converted tabs to 2 space, lemme rever that.

File: DEVELOPERS.md (hunk 1 above)
Comment (AnEmortalKid) [reply DEVELOPERS.md:8]: Yep. Since I have a windows machine, I'll work on this then!

File: src/parser.djs [hunk 3]
```diff
@@ -1516,8 +1529,8 @@ such parseFunctionDeclaration much content isClassStatic
                 throw err
             wow
 
-            very newlineStart is content giv content dose startsWith with '\n'
-            rly newlineStart
+            very textContent is content giv content
+            rly plz startsWithNewLine with textContent thx
```
Comment (AnEmortalKid) [inline src/parser.djs:1533]:
```
$ cat snippet.djs
rly plz startsWithNewLine with textContent thx
wow

$ dogescript snippet.djs  --beautify
if (startsWithNewLine(textContent)) {}
```

you can invoke function calls by making use of  `thx` to terminate them in certain spots. However if you'd like me to keep the syntax consistent or if it helps to decipher by just doing newLineStart, I don't mind changing that.

Comment (vpzomtrrfrt) [reply src/parser.djs:1533]: at the time I wrote this code I don't think that worked but it was probably fixed in 2.4

looks fine

File: src/parser.djs [hunk 4]
```diff
@@ -68,6 +69,16 @@ such isInlineWhitespace much chr
     wow
 wow result
 
+quiet
+  Determines if the first character of the content is a new line character
+  
+  @return true if the content starts with a new line character, false otherwise
+loud
+such startsWithNewLine much content
```
Comment (vpzomtrrfrt) [inline src/parser.djs:77]: I'm not sure \r should count as a newline, are there issues with just including it as inlineWhitespace?

**Review by vpzomtrrfrt — CHANGES_REQUESTED** [review]

File: src/parser.djs (hunk 4 above)
Comment (AnEmortalKid) [reply src/parser.djs:77]: No there weren't! It ends up correctly consuming the `\'r'` at the end of lines now as well and not throwing parse errors!

File: src/parser.djs [hunk 5]
```diff
@@ -135,7 +136,7 @@ such ifSkippedComment much content
     rly shhStart
         very done is false
         many content giv length bigger 0 and done not true
-            rly content levl 0 is '\n'
+            rly content levl 0 is '\n' or content levl 0 is '\r'
```
Comment (vpzomtrrfrt) [inline src/parser.djs:139]: this also seems unnecessary since the \r can just be part of the comment

**Review by vpzomtrrfrt — APPROVED** [review]

**Pull request merged**